File: Sistema_de_Salarios/Salarios_jv/src/Teste.java
import java.util.InputMismatchException;
import java.util.Scanner;
public class Teste {
    Funcionario[] funcionarios;
    Scanner scanner;

    //CONSTRUTOR
    public Teste(int n_funcionarios){
        funcionarios = new Funcionario[n_funcionarios];
        scanner = new Scanner(System.in);
    }

    public void leitura_teste(int n_gerentes, int n_vendedores, int n_assistentes){

        String buffer_nome;
        String buffer_CPF;
        double buffer_comissao;
        boolean flag_leitura;

        for(int i = 0; i < n_gerentes; i++){
            flag_leitura = false;
            while(!flag_leitura){
                try{
                    System.out.print("Insira o nome e o Cpf do gerente a ser incluído, um em cada linha\n");
                    buffer_nome = scanner.nextLine();
                    buffer_CPF = scanner.nextLine();
                    funcionarios[i] = new Gerente(buffer_nome, buffer_CPF);
                    flag_leitura = true;
                }catch (CPF_Invalido e){
                    System.out.println("CPF inserido inválido, tente outro CPF.");
                    scanner.nextLine();
                }catch (InputMismatchException e){
                    System.out.println("Input inserido inválido, tente novamente.");
                    scanner.nextLine();
                }
            }
        }


        for(int i = n_gerentes; i < n_gerentes + n_assistentes; i++){
            flag_leitura = false;
            while(!flag_leitura){
                try{
                    System.out.print("Insira o nome e o Cpf do assistente a ser incluído, um em cada linha\n");
                    buffer_nome = scanner.nextLine();
                    buffer_CPF = scanner.nextLine();
                    funcionarios[i] = new Assistente(buffer_nome, buffer_CPF);
                    flag_leitura = true;
                }catch (CPF_Invalido e){
                    System.out.println("CPF inserido inválido, tente outro CPF.");
                    scanner.nextLine();
                }catch (InputMismatchException e){
                    System.out.println("Input inserido inválido, tente novamente.");
                    scanner.nextLine();
                }
            }
        }

        for(int i = (n_gerentes + n_assistentes); i < (n_gerentes + n_assistentes + n_vendedores); i++) {
            flag_leitura = false;
            while (!flag_leitura) {
                try {
                    System.out.print("Insira o nome, o Cpf e a comissão do vendedor a ser incluído, um em cada linha\n");
                    buffer_nome = scanner.nextLine();
                    buffer_CPF = scanner.nextLine();
                    buffer_comissao = scanner.nextDouble();
                    funcionarios[i] = new Vendedor(buffer_nome, buffer_CPF, buffer_comissao);
                    flag_leitura = true;
                }catch (CPF_Invalido e){
                    System.out.println("CPF inserido inválido, tente outro CPF.");
                    scanner.nextLine();
                }catch (InputMismatchException e){
                    System.out.println("Input inserido inválido, tente novamente.f");
                    scanner.nextLine();
                }
            }
        }
        scanner.close();
    }

    public void print_folha_salarial(){
        int contador = 0 ;
        for(Funcionario f : funcionarios){
            System.out.printf("Funcionário %d :\n", contador);
            contador++;
            System.out.println(f.toString());
        }
    }

    public static int teste_leitura_nextInt(Scanner scanner1){
        int returned_value = 0;
        boolean control_flag = false;
        while (!control_flag){
            try{
                returned_value = scanner1.nextInt();
                control_flag = true;
            }catch(InputMismatchException e){
                System.out.println("Número inserido inválido, tente novamente");
                scanner1.nextLine();
            }
        }
        return returned_value;
    }

}

File: Sistema_de_Salarios/Salarios_py/Teste.py
from Funcionario import Gerente, Assistente, Vendedor


class Teste:
    # CONSTRUTOR
    def __init__(self, n_funcionarios):
        self.funcionario = [None] * n_funcionarios

    # LEITURA
    def leitura(self, n_gerentes, n_vendedores, n_assistentes):
        contador = 0

        # leitura dos gerentes:
        flag_leitura = False
        while not flag_leitura and contador < n_gerentes:
            try:
                buffer_nome = input("Insira o nome do gerente a ser incluído.\n")
                buffer_cpf = input("Insira o cpf do gerente a ser incluído.\n")
                self.funcionario[contador] = Gerente(buffer_nome, buffer_cpf)
                flag_leitura = True
                contador = contador + 1
            except Exception as e:
                print(f"Erro {contador}:", str(e), "Tente novamente \n")

        # leitura dos vendedores:
        flag_leitura = False
        while not flag_leitura and contador < (n_gerentes + n_vendedores):
            try:
                buffer_nome = input("Insira o nome do vendedor a ser incluído.\n")
                buffer_cpf = input("Insira o cpf do vendedor a ser incluído.\n")
                buffer_comissao = float(input("Insira a comissão do vendendor a ser incluído\n"))
                self.funcionario[contador] = Vendedor(buffer_nome, buffer_cpf, buffer_comissao)
                flag_leitura = True
                contador = contador + 1
            except Exception as e:
                print(f"Erro {contador}:", str(e), "Tente novamente\n")

        # leitura dos assistentes:
        flag_leitura = False
        while not flag_leitura and contador < (n_gerentes + n_vendedores + n_assistentes):
            try:
                buffer_nome = input("Insira o nome do assistente a ser incluído.\n")
                buffer_cpf = input("Insira o cpf do assistente a ser incluído.\n")
                self.funcionario[contador] = Assistente(buffer_nome, buffer_cpf)
                flag_leitura = True
                contador = contador + 1
            except Exception as e:
                print(f"Erro {contador}:", str(e), "Tente novamente\n")

    # Impressão
    def print_folha_salarial(self):
        contador = 0
        for f in self.funcionario:
            print(f"funcionário {contador}:")
            print(str(f))
            contador = contador + 1
